pkg/controllers/jobset_controller.go

```diff
@@ -178,14 +177,29 @@ func SetupIndexes(indexer client.FieldIndexer) error {
 func (r *JobSetReconciler) getChildJobs(ctx context.Context, js *jobset.JobSet) (*childJobs, error) {
 	log := ctrl.LoggerFrom(ctx)
 
+	// Set up the success policy's job selector.
+	labelMap, err := metav1.LabelSelectorAsMap(js.Spec.SuccessPolicy.JobSelector)
+	if err != nil {
+		return nil, err
+	}
+	jobSelector := labels.Set(labelMap).AsSelector()
+
 	// Get all active jobs owned by JobSet.
 	var childJobList batchv1.JobList
 	if err := r.List(ctx, &childJobList, client.InNamespace(js.Namespace), client.MatchingFields{jobOwnerKey: js.Name}); err != nil {
 		return nil, err
 	}
+
 	// Categorize each job into a bucket: active, successful, failed, or delete.
 	ownedJobs := childJobs{}
 	for i, job := range childJobList.Items {
+
+		// Check if this job matches the success policy's job selector.
+		matchingSuccessSelector := jobSelector.Matches(labels.Set(job.Labels))
+		if matchingSuccessSelector {
+			ownedJobs.matchingSuccessSelector += 1
+		}
```

Comment:
What if childJobList doesn't contain all the jobs that JobSet is supposed to have? for this reason, we need to calculate the number based on the JobSet spec.

Reply:
I'm not sure if it's possible to generically apply the success policy's jobSelector against the JobSet spec (replicatedJob templates) to calculate this, since the user won't be able to use any of the labels added by the JobSet controller that aren't part of the spec (`jobset.sigs.k8s.io/job-index`, etc.). 

So if we want the success policy's jobSelector to be able to utilize the labels added by the jobset controller (I think we do, especially `jobset.sigs.k8s.io/job-index`), then I think we will need think of another approach. What do you think?

Reply:
Maybe the jobset controller can add these labels/annotations to the replicatedJob template as well on the first reconciliation? That way the success policy jobSelector can just look at the spec in order to calculate the number of jobs are required to succeed.

Reply:
hmm, yeah, computing that will be expensive. I guess the best path forward for now is to change the api from a selector to a list of replicatedJobNames as you suggested on the issue. We can expand the api to add a selector in the future if we want to.

Reply:
Alright I've updated the PR to implement the success policy API as described in https://github.com/kubernetes-sigs/jobset/issues/82#issuecomment-1536839914 and this is ready for another look